swupd/create_manifests.go

```diff
@@ -136,14 +136,22 @@ func initBundles(ui UpdateInfo, c config) ([]*Manifest, error) {
 		go bundleWorker()
 	}
 
-	for _, bn := range ui.bundles {
-		select {
-		case bundleChan <- bn:
-		case err = <-errorChan:
-			// break as soon as we see a failure
-			break
+	// Generator
+	var err error
```

Comment:
I don't think this is the right way to do this, I think a simpler method could be used - **as long as the intended result is to quit execution of mixer if something fails here:**
```
var err error
// Exit earliy if we get an error
go func() {
	for e := range errorChan {
		fmt.Println(e)
		os.Exit(1)
	}
}()

for i := 0; i < workers; i++ {
	bundleChan <- i
}
close(bundleChan)
wg.Wait()
close(errorChan) // So we don't deadlock in error goroutine if we didn't get any errors
```
Even in the goroutine you start for adding bundles to bundleChan, it will never hit the error case because there are always goroutines available to pick up a bundle.

Reply:
It does work, the routines stop getting bundles once one of them sends an error. The problem before was that break was a no-op since it was breaking the select instead of the for loop. But this code works as intended.

Reply:
Ugh, you're right https://tip.golang.org/ref/spec#Break_statements
It never actually breaks out of the for in the original code. Return works since you're spawning off a routine, but perhaps we could do it without a waitgroup or extra goroutine if we follow this code ^ and do 
```
BUNDLE_LOOP:
	for _, bn := range ui.bundles {
		select {
		case bundleChan <- bn:
		case err = <- errorChan:
			break BUNDLE_LOOP
		}
	}
```